File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    let answer = 0;

    nums.sort((a, b) => a - b);

    const max =
        nums[nums.length - 1] +
        nums[nums.length - 2] +
        nums[nums.length - 3];

    // 에라토스테네스의 체
    const isPrime = new Array(max + 1).fill(true);
    isPrime[0] = false;
    isPrime[1] = false;

    for (let i = 2; i * i <= max; i++) {
        if (!isPrime[i]) continue;

        for (let j = i * i; j <= max; j += i) {
            isPrime[j] = false;
        }
    }

    // 서로 다른 3개의 수를 선택
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            for (let k = j + 1; k < nums.length; k++) {
                if (isPrime[nums[i] + nums[j] + nums[k]]) {
                    answer++;
                }
            }
        }
    }

    return answer;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MakePrimeNumber {
	public static void main(String[] args) {
		Solution s = new Solution();
		
		int[] nums1 = {1,2,3,4};
		System.out.println("result : " + s.solution(nums1));
		
		int[] nums2 = {1,2,7,6,4};
		System.out.println("result : " + s.solution(nums2));
	}
}

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        Arrays.sort(nums);
        int max = nums[nums.length-1] + nums[nums.length - 2] + nums[nums.length - 3];
        
        List<Boolean> list = new ArrayList<Boolean>();
        list.add(false); list.add(false);
        for (int i = 2; i <= max; i++) list.add(true);
        
        for (int i = 2; i*i <= max; i++) {
        	for (int j = i*i; j <= max; j += i) {
        		if(list.get(j)) list.set(j, false);
			}
		}
        
        for (int i = 0; i < nums.length; i++) {
			for (int j = i+1; j < nums.length; j++) {
				for (int k = j+1; k < nums.length; k++) {
					answer += list.get(nums[i] + nums[j] + nums[k]) ? 1 : 0;
				}
			}
		}

        return answer;
    }
}